cilium-dbg/cmd/encrypt_status.go

```diff
@@ -57,71 +58,93 @@ func init() {
-	case models.EncryptionStatusModeWireguard:
-		return dumpWireGuardStatus(enc), nil
+	result.Ipsec, err = dumpIPsecStatus()
+	if err != nil {
+		return result, err
 	}
-	return models.EncryptionStatus{Mode: models.EncryptionStatusModeDisabled}, nil
-}
 
-func dumpIPsecStatus() (models.EncryptionStatus, error) {
-	status := models.EncryptionStatus{
-		Mode:  models.EncryptionStatusModeIPsec,
-		Ipsec: &models.IPsecStatus{},
+	result.Wireguard, err = dumpWireGuardStatus()
+	if err != nil {
+		return result, err
 	}
+
+	return result, nil
+}
+
+func dumpIPsecStatus() (*models.IPsecStatus, error) {
 	xfrmStates, err := safenetlink.XfrmStateList(netlink.FAMILY_ALL)
 	if err != nil {
-		return models.EncryptionStatus{}, fmt.Errorf("cannot get xfrm state: %w", err)
+		return nil, fmt.Errorf("cannot get xfrm state: %w", err)
 	}
+
 	keys, err := ipsec.CountUniqueIPsecKeys(xfrmStates)
 	if err != nil {
-		return models.EncryptionStatus{}, fmt.Errorf("error counting IPsec keys: %w", err)
+		return nil, fmt.Errorf("error counting IPsec keys: %w", err)
+	}
+
+	// no ipsec state installed
+	if keys == 0 {
+		return nil, nil
 	}
-	status.Ipsec.KeysInUse = int64(keys)
-	decryptInts, err := getDecryptionInterfaces()
+
+	var result models.IPsecStatus
+
+	result.KeysInUse = int64(keys)
+
+	result.DecryptInterfaces, err = getDecryptionInterfaces()
 	if err != nil {
-		return models.EncryptionStatus{}, fmt.Errorf("error getting IPsec decryption interfaces: %w", err)
+		return nil, fmt.Errorf("error getting IPsec decryption interfaces: %w", err)
 	}
-	status.Ipsec.DecryptInterfaces = decryptInts
-	seqNum, err := maxSequenceNumber()
+
+	result.MaxSeqNumber, err = maxSequenceNumber()
 	if err != nil {
-		return models.EncryptionStatus{}, fmt.Errorf("error getting IPsec max sequence number: %w", err)
+		return nil, fmt.Errorf("error getting IPsec max sequence number: %w", err)
 	}
-	status.Ipsec.MaxSeqNumber = seqNum
+
 	errCount, errMap, err := getXfrmStats("")
 	if err != nil {
-		return models.EncryptionStatus{}, fmt.Errorf("error getting xfrm stats: %w", err)
+		return nil, fmt.Errorf("error getting xfrm stats: %w", err)
 	}
-	status.Ipsec.ErrorCount = errCount
-	status.Ipsec.XfrmErrors = errMap
-	return status, nil
+
+	result.ErrorCount = errCount
+	result.XfrmErrors = errMap
+	return &result, nil
 }
 
-func dumpWireGuardStatus(p *models.EncryptionStatus) models.EncryptionStatus {
-	status := models.EncryptionStatus{
-		Mode: models.EncryptionStatusModeWireguard,
-		Wireguard: &models.WireguardStatus{
-			Interfaces: make([]*models.WireguardInterface, 0),
-		},
-	}
-	for _, wg := range p.Wireguard.Interfaces {
-		status.Wireguard.Interfaces = append(status.Wireguard.Interfaces, &models.WireguardInterface{
-			Name:      wg.Name,
-			PublicKey: wg.PublicKey,
-			PeerCount: wg.PeerCount,
+func dumpWireGuardStatus() (*models.WireguardStatus, error) {
+	wgClient, err := wgctrl.New()
+	if err != nil {
+		return nil, err
+	}
+
+	devices, err := wgClient.Devices()
```

Comment:
Nice, but I'd use directly the `types.IfaceName` value (cilium_wg0) as in the `WireguardAgent.Status` function [here](https://github.com/cilium/cilium/blob/main/pkg/wireguard/agent/agent.go#L840). The aim in this case (fallback, no reply from agent) is exactly to mimic that. If enabled we have at max 1 wireguard interface. I'm afraid with `wgClient.Devices()` we would pick also non-Cilium related ones, if any 😃

Reply:
yes! that makes sense. just like we did in the ipsec (we filter our states by the value we know the agent added to the states it manages) - this way we can avoid returning state that was added by non-cilium things on the node
i'll implement that